Java-Basics.md

```diff
@@ -0,0 +1,442 @@
+	}
+}
+
+ElectricCar modelS = new ElectricCar("Model S","Tesla");
+// prints Tesla
+System.out.println(modelS.getName());
+// prints Charging ...
+modelS.charge();
+```
+
+See here that the class `ElecticCar` inherits or `extends` the public methods from `Car` class, as well as has its own methods and properties. Cool way to pass on information!
+
+Also notice the usage of [super]() keyword here. Since our `Car` class had a constructor, so we have to initialize that constructor from the child class as well. We do that using the `super` keyword. Read more about [Inheritance here](#TODO).
+
+### Interfaces
+
+But wait, my manager has given me a set of strict specifications every class to create, but has told me you can implement it in whichever way you want. Incidentally, Java has a nifty feature of Interfaces which does exactly that!
+
+```java
+interface Car {
+	public String getName();
+	public String getManufacturerName();
+}
+
+class ElectricCar implements Car {
+	private String name;
+	private String manufacturerName;
+
+	public ElectricCar(String name, String man) {
+	  this.name = name;
+	  this.manufacturerName = man;
+    }
+
+	public void charge() {
+	 System.out.println("Charging ...");
+	}
+
+	// Getter method
+	public String getName() {
+		return name;
+	}
+  	// Getter method
+	public String getManufacturerName() {
+		return manufacturerName;
+	}
+}
+```
+
+So interface basically bounds you to a contract to follow, where you must _implement_ all the methods. If you don't, the compiler will complain! Know more about the [awesome power of Inheritance here](#TODO).
+
+## Basic Operations
+
+Java supports the following operations on variables:
+
+* Arithmetic : `Addition(+)`, `Subtraction(-)`, `Multiplication(*)`, `Division(/)`, `Modulus(%)`,`Increment(++)`, `Decrement(--)`.
+* String concatenation:  `+` can be used for String concatenation but subtraction `-` on a String is not a valid operation.
+* Relational: `Equal to(==)`, `Not Equal to (!=)`, `Greater than(>)`, `Less than(<)`, `Greater than or equal to(>=)`, `Less than or equal to(<=)`, 
+* Bitwise: `Bitwise And(&)`, `Bitwise Or(|)`, `Bitwise XOR(^)`, `Bitwise Compliment(~)`, `Left shift(<<)`, `Right Shift (>>)`, `Zero fill right shift (>>>)`
+* Logical: `Logical And (&&)`, `Logical Or(||)`, `Logical Not (!)`
+* Assignment: `=`, `+=`, `-=`, `*=`, `/=`, `%=`, `<<=`, `>>=`, `&=`, `^=`, `|=`
+* Others: `Conditional/Ternary(?:)`, `instanceof`
+
+While most of the operations are self explanatory, the Conditional (Ternary) Operator works as follows: 
+`expression that results in boolean output ? return this value if true : return this value if false`
+
+For e.g: 
+```java
+int x = 10;
+int y = (x == 10) ? 5 : 9; <-- y will equal 5 since the expression x == 10 evaluates to true
+```
+
+
+###### The `instanceof` operator
+This is a special operator that allows the developer to check whether or not a particular object is in the hierarchy of particular Class/Interface.
```

Comment:
Not clear what you are saying here, at least to a beginner.

An object can be part of the hierarchy, irrespective of where in hierarchy it is. It could be above it or below it.

But, only `vehicle instanceof Car` would return `true`. Not `Car instanceof vehicle`.